app/models/results/response_tree_builder.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Results
+  # Builds a response tree from a form object
+  class ResponseTreeBuilder
+    def initialize(form)
+      @form = form
+    end
+
+    def build
+      @rn = AnswerGroup.new
+      add_children(@form.root_group.children, @rn)
+      @rn
+    end
+
+    def add_children(form_node, response_node)
+      form_node.each do |q|
+        if q.class == QingGroup && q.repeatable?
+          response_node.children << AnswerGroupSet.new(questioning_id: q.id, new_rank: q.rank)
+          add_repeat_groups(q, response_node.children.last)
```

Comment:
I think it's better to assign the AnswerGroupSet to a var and use that in both the above lines than to use `children.last`. The latter makes me a bit nervous. Sometimes methods like `children` will sort things or do other magic things. Better not to trifle with it.